### PR TITLE
Added warning to invalid countmax in Monsters::loadLootItem

### PR DESCRIPTION
**Pull request by EPuncker**

### CODE REVIEW

File: src/monsters.cpp [hunk 1]
```diff
@@ -1385,6 +1385,11 @@ bool Monsters::loadLootItem(const pugi::xml_node& node, LootBlock& lootBlock)
 	}
 
 	if ((attr = node.attribute("countmax"))) {
+		int32_t lootCountmax = pugi::cast<int32_t>(attr.value());
+		if (lootCountmax > 100) {
```
Comment (marmichalski) [inline src/monsters.cpp:1389]: what about negative value? 🌞

File: src/monsters.cpp [hunk 2]
```diff
@@ -1385,6 +1385,11 @@ bool Monsters::loadLootItem(const pugi::xml_node& node, LootBlock& lootBlock)
...
+			std::cout << "[Warning - Monsters::loadMonster] Invalid \"countmax\" "<< lootCountmax <<" used for loot, the max allowed value is 100. " << std::endl;
+			return false;
+		}
 		lootBlock.countmax = std::max<int32_t>(1, pugi::cast<int32_t>(attr.value()));
```
Comment (marmichalski) [inline src/monsters.cpp:1393]: you can reuse the variable `lootCountmax` here, which should prolly be written as `lootCountMax`

File: src/monsters.cpp (hunk 1 above)
Comment (MillhioreBT) [reply src/monsters.cpp:1389]: There is no need to handle negative values ​​if the value is extracted and passed through the function:
```c++
std::max<int32_t>(1, pugi::cast<int32_t>(attr.value()));
```
first.

Comment (marmichalski) [reply src/monsters.cpp:1389]: Then why not use `std::clamp` and do not warn at all? I don't see a reason to warn about too high value, but not warn about too low value. It should be either both or none at all. 🤷

Comment (DSpeichert) [reply src/monsters.cpp:1389]: A warning could be useful although it's quite unlikely that someone will try to put in a negative value for loot.

**Review by DSpeichert — DISMISSED** [review]

File: src/monsters.cpp (hunk 1 above)
Comment (EPuncker) [reply src/monsters.cpp]: something like this?
```
		if (lootCountMax > 100 || lootCountMax < 1) {
			std::cout << "[Warning - Monsters::loadMonster] Value for \"countmax\" "<< lootCountMax <<" out of bounds. " << std::endl;
			return false;
		}
```

**Review by DSpeichert — APPROVED** [review]

**Comment by soul4soul** [comment]
How about we merge #2875 instead?

**Comment by EPuncker** [comment]
> How about we merge #2875 instead?

don't worry, this is just a temporary measure while #2906 doesn't get in

**Pull request merged**